tests/tom-swifty-test/SwiftReflector/NewClassCompilerTests.cs

```diff
@@ -1533,6 +1533,96 @@ public class DisposeMeThanks {
 
 			TestRunning.TestAndExecute (swiftCode, callingCode, "SwiftObject handle is IntPtr.Zero, likely because it was disposed.\n");
 		}
+
+		[Test]
+		public void SandwichTest ()
+		{
+			var swiftCode = @"
+public protocol Filling {
+    var stuff: String { get }
+}
+
+public protocol Bread {
+    var name: String { get }
+    var sliced: Bool { get }
+}
+
+public struct Rye : Bread {
+    public init () { }
+    public var name:String {
+        get { return ""rye"" }
+    }
+    public var sliced:Bool {
+        get { return true }
+    }
+}
+
+public struct Ham : Filling {
+    public init () { }
+    public var stuff: String {
+        get { return ""ham"" }
+    }
+}
+
+public enum LettuceKind {
+    case IceBerg, Romaine, Bib
+}
+
+public struct Lettuce : Filling {
+    private var kind: LettuceKind
+    public init (of: LettuceKind) {
+        kind = of
+    }
+    public var stuff: String {
+        get {
+            return ""\(kindToString ()) lettuce""
+        }
+    }
+    
+    private func kindToString () -> String {
+        switch kind {
+        case .IceBerg: return ""iceberg""
+        case .Romaine: return ""romaine""
+        case .Bib: return ""bib""
+        }
+    }
+}
+
+public func printSandwich (of: Bread, with: Filling) {
+    print (""\(with.stuff) on \(of.name)"")
+}
+";
+
+			var printer = CSFunctionCall.ConsoleWriteLine (CSConstant.Val ("hi"));
+			var callingCode = CSCodeBlock.Create (printer);
+
+			TestRunning.TestAndExecute (swiftCode, callingCode, "hi\n");
```

Comment:
what does this test do ?

Reply:
it generates code that I needed for a sample and for (future) debugging. It is benign.